kusto/kusto.go

```diff
@@ -36,13 +36,17 @@ type Authorization struct {
 	// Config provides the authorizer's config that can create the authorizer. We recommending setting
 	// this instead of Authorizer, as we will automatically set the Resource ID with the endpoint passed.
 	Config auth.AuthorizerConfig
+	// Token provider that can be used to get the access token.
+	tokenProvider TokenProvider
 }
 
 // Validate validates the Authorization object against the endpoint an preps it for use.
 // For internal use only.
 func (a *Authorization) Validate(endpoint string) error {
 	const rescField = "Resource"
-
+	if a.tokenProvider != (TokenProvider{}) {
```

Comment:
Q: What does this check do actually ?

Reply:
@yogilad Since we cannot compare the structs with nil in Go. We have to compare it with an empty struct of same type to check if its unset.

Reply:
Maybe hold a pointer that can be nil?

Reply:
@AsafMah Yeah agree to it. But we needed a solid reason to go for a pointer-based approach.   Instead, we will be using an isInitialized flag in the struct to check if the struct is created or not...

Reply:
one reason could be, if we make tokenProvider of pointer type, the reference would be used, hence making it more reliable in cases where any modifications occur in tokenProvider.